WeatherToday/Sources/Data/NetworkAPI/Core/RequestConfiguration.swift

```diff
@@ -2,38 +2,55 @@
 //  RequestConfiguration.swift
 //
 
-import Alamofire
 import Foundation
+import Moya
 
-protocol RequestConfiguration {
-
-    var baseURL: String { get }
-
-    var endpoint: String { get }
-
-    var method: HTTPMethod { get }
-
-    var url: URLConvertible { get }
-
-    var parameters: Parameters? { get }
+enum RequestConfiguration {
+    case weather(cityName: String)
+}
 
-    var encoding: ParameterEncoding { get }
+extension RequestConfiguration: TargetType {
 
-    var headers: HTTPHeaders? { get }
+    var baseURL: URL { URL(string: Configuration.baseWeatherURL)! }
 
-    var interceptor: RequestInterceptor? { get }
-}
+    var path: String {
+        switch self {
+        case .weather:
+            return "data/2.5/weather"
+        }
+    }
 
-extension RequestConfiguration {
+    var method: Moya.Method {
+        switch self {
+        case .weather:
+            return .get
+        }
+    }
 
-    var url: URLConvertible {
-        let url = URL(string: baseURL)?.appendingPathComponent(endpoint)
-        return url?.absoluteString ?? "\(baseURL)\(endpoint)"
+    var task: Moya.Task {
+
+        switch self {
+        case .weather(let cityName):
+                return Task.requestParameters(
+                    parameters: [
+                        "q": cityName,
+                        "appid": Configuration.weatherApiKey,
+                        "units": "metric"
+                    ],
+                    encoding: URLEncoding.default
+                )
+        }
     }
 
-    var parameters: Parameters? { nil }
+    var headers: [String : String]? { ["Content-Type": "application/json"] }
 
-    var headers: HTTPHeaders? { nil }
+    // Optional Stub
+    var sampleData: Data {
+        return Data()
+    }
 
-    var interceptor: RequestInterceptor? { nil }
+    // Optional Stub
```

Comment:
This is not optional since we declare some real value to it, I suggest removing this comment.

Reply:
By the comment I want to say that : these stubs can be skipped, actually I try to pretend like this is for iOS Template.

Reply:
I don't think we need to add comment as a documentation. When we use a library, the document is available on the library's repository. Adding library's comment in the project will lead to: - bloated code base & - need to revise comment if the library document changes.

We assume reader of this code knows Moya (because we pick Moya as part of the project) so we should not include the comment.